tests/components/hassio/test_discovery.py

```diff
@@ -2,12 +2,21 @@
 from http import HTTPStatus
 from unittest.mock import Mock, patch
 
+import pytest
+
 from homeassistant.components.hassio import HassioServiceInfo
 from homeassistant.components.hassio.handler import HassioAPIError
 from homeassistant.const import EVENT_HOMEASSISTANT_START
 from homeassistant.setup import async_setup_component
 
 
+@pytest.fixture(autouse=True)
+def mock_repairs(hass):
+    """Mock the repairs integration setup."""
+    with patch("homeassistant.components.repairs.websocket_api.async_setup"):
```

Comment:
Why do the tests fail without this?

Reply:
The tests load MQTT, and when repairs is added as dependency repairs fails to load. This is what's is patched.

Reply:
This happens without patch:

```log
FAILED tests/components/hassio/test_discovery.py::test_hassio_discovery_startup[pyloop] - ModuleNotFoundError: No module named 'homeassistant.components.requests'
FAILED tests/components/hassio/test_discovery.py::test_hassio_discovery_startup_done[pyloop] - AssertionError: assert False
FAILED tests/components/hassio/test_discovery.py::test_hassio_discovery_startup_done[pyloop] - homeassistant.exceptions.DependencyError: (DependencyError(...), 'Could not setup dependencies: repairs')
FAILED tests/components/hassio/test_discovery.py::test_hassio_discovery_webhook[pyloop] - AssertionError: assert 500 == <HTTPStatus.OK: 200>

Results (0.73s):
       3 failed
         - tests/components/hassio/test_discovery.py:11 test_hassio_discovery_startup[pyloop]
         - tests/components/hassio/test_discovery.py:66 test_hassio_discovery_startup_done[pyloop]
         - tests/components/hassio/test_discovery.py:129 test_hassio_discovery_webhook[pyloop]
```

Reply:
It's not clear to me why the tests fail.

The tests should probably be rewritten to not use the mqtt integration but create a mock integration with a config flow similar to this test:
https://github.com/home-assistant/core/blob/1aa0e64354d1dffd5e948da803e4d379ee39443c/tests/test_config_entries.py#L678-L698

Reply:
That sounds like a better idea.

Reply:
Before we rewite tests:
The tests wants to test if hassio discovery works. MQTT is a built-in discoverable service, So it is not that stange to use this. What goes wrong is that `hass.http.register_view()` is called when initializing `repairs`. This fails, so I mocked the setup function that is calling this. We also mock `aioclient`, and `hassio_client`. So it the current tests seem fine to me, but they should not fail when we add the `repairs` integration to the `mqtt` dependencies. That is what I am trying to solve here.

Reply:
The hassio integration doesn't depend on the mqtt integration so the tests for the hassio integration shouldn't use anythong from the mqtt integration.

Why is it wrong to call `hass.http.register_view()` when initializing `repairs`?

Reply:
> Why is it wrong to call `hass.http.register_view()` when initializing `repairs`?

I don't know, but it fails.

Reply:
We need to know why it fails before we patch it. But I suggest rewriting these tests instead.

Reply:
The reason the http stuff fails is because we end up with the `"Cannot register a resource into frozen router."` error if not being very careful when mixing ws + http APIs in tests.